fn/mvar.go

```diff
@@ -0,0 +1,205 @@
+package fn
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+// MVar[A any] is a structure that is designed to store a single value in an API
```

Comment:
I think could use some more detail in the commit body re the rationale/usage for such a variable. Eg: does the `M` prefix have any intended semantic meaning?

Reply:
Would you rather have it in the commit body or just in the godoc itself?